core/src/main/java/fj/Equal.java

```diff
@@ -93,15 +85,7 @@ public Boolean f(final A a1) {
    *         equality.
    */
   public static <A> Equal<A> anyEqual() {
-    return new Equal<A>(new F<A, F<A, Boolean>>() {
-      public F<A, Boolean> f(final A a1) {
-        return new F<A, Boolean>() {
-          public Boolean f(final A a2) {
-            return a1.equals(a2);
-          }
-        };
-      }
-    });
+    return new Equal<A>(a1 -> a2 -> a1.equals(a2));
```

Comment:
Did not try this myself, but I believe you should use `Equal.equal(a1 -> a2 -> a1.equals(a2));` to take advantage of the type inference.  This applies in lots of case below where the type parameter gets more complex.